File: py/insertionsort.py
def insertionsort():
	i=1;
	while i!=n:
		temp= a[i];
		j=i-1;
		while(j!=-1 and temp<=a[j]):
			a[j+1]= a[j];
			j=j-1;
		a[j+1]= temp;
		i=i+1;
	return a;

n= input('Enter the no of elements in array: ');
a=[];
for i in range(n):
	b= input();
	a.append(b);
insertionsort();
print 'The sorted array: ';
print a;

File: algo/c/insertion_sort.c
#include <stdio.h>
int a[20];

void selectionsort(int n)
{
	int i,j, temp, loc;
	for(i=0;i<n-1;i++)
	{
		loc=i;
		for(j=i+1;j<n;j++)
		{
			if(a[j]<a[loc])
			{
				loc=j;
			}
		}
		if(loc!=i)
		{
			temp=a[i];
			a[i]= a[i+1];
			a[i+1]=temp;
		}
	}
}

int main()
{
	int i,n;
	printf("Enter the no of elements to be sorted");
	scanf("%d",&n);
	printf("Enter the array to be sorted");
	for(i=1;i<=n;i++)
	{
		scanf("%d", &a[i]);
	}
	selectionsort(n);
	for(i=1;i<=n;i++)
	{
		printf("%d ", a[i]);
	}
	return 0;
}

